plumbing/object/tree.go

```diff
@@ -279,7 +302,15 @@ func (t *Tree) Encode(o plumbing.EncodedObject) (err error) {
 	}
 
 	defer ioutil.CheckClose(w, &err)
+
+	if !sort.IsSorted(TreeEntrySorter(t.Entries)) {
+		return errors.New("entries in tree is not sorted")
```

Comment:
Let's make this a named error, moving it together with the other `Err...` vars at the top of the file.